File: _CONTAINER/_BY_MONTH/_Feb/Thesis-physics-master/runtime/engine/math.cpp

#include "math.h"

/**
 * @param {number} x
 * @return {number}
 */
float smash::math::square(float x) {
  return x * x;
};


/**
 * @param {number} x1
 * @param {number} y1
 * @param {number} z1
 * @param {number} x2
 * @param {number} y2
 * @param {number} z2
 * @return {number}
 */
float smash::math::vectorDistance(float x1, float y1, float z1, float x2,
    float y2, float z2) {
  return sqrt(smash::math::square(x1 - x2) +
      smash::math::square(y1 - y2) +
      smash::math::square(z1 - z2));
};


/**
 * @param {number} x
 * @param {number} y
 * @param {number} z
 * @return {number}
 */
float smash::math::vectorLength(float x, float y, float z) {
  return sqrt(smash::math::square(x) +
      smash::math::square(y) +
      smash::math::square(z));
};

/**
 * @param {number} x
 * @param {number} y
 * @param {number} z
 * @return {number}
 */
float smash::math::dot(float x1, float y1, float z1, float x2, float y2,
    float z2) {
  return x1 * x2 + y1 * y2 + z1 * z2;
};


/**
 * @param {!smash.Sphere} sphere1
 * @param {!smash.Sphere} sphere2
 * @return {boolean}
 */
bool smash::math::checkCollidingSpheres(smash::Sphere* sphere1,
    smash::Sphere* sphere2) {
  return smash::math::vectorDistance(
      sphere1->positionX, sphere1->positionY, sphere1->positionZ,
      sphere2->positionX, sphere2->positionY, sphere2->positionZ) <
      sphere1->radius + sphere2->radius;
};

File: _BY_MONTH/_Feb/Thesis-physics-master/browser/engine/math.js
/**
 * @fileoverview Math utils.
 * @author sebastian.poreba@gmail.com (Sebastian Poreba)
 */

goog.provide('smash.math');


/**
 * @param {number} x
 * @return {number}
 */
smash.math.square = function(x) {
  return x * x;
};


/**
 * @param {number} x1
 * @param {number} y1
 * @param {number} z1
 * @param {number} x2
 * @param {number} y2
 * @param {number} z2
 * @return {number}
 */
smash.math.vectorDistance = function(x1, y1, z1, x2, y2, z2) {
  return Math.sqrt(smash.math.square(x1 - x2) +
      smash.math.square(y1 - y2) +
      smash.math.square(z1 - z2));
};


/**
 * @param {number} x
 * @param {number} y
 * @param {number} z
 * @return {number}
 */
smash.math.vectorLength = function(x, y, z) {
  return Math.sqrt(smash.math.square(x) +
      smash.math.square(y) +
      smash.math.square(z));
};


/**
 * @param {!smash.Sphere} sphere1
 * @param {!smash.Sphere} sphere2
 * @return {boolean}
 */
smash.math.checkCollidingSpheres = function(sphere1, sphere2) {
  return smash.math.vectorDistance(
      sphere1.positionX, sphere1.positionY, sphere1.positionZ,
      sphere2.positionX, sphere2.positionY, sphere2.positionZ) <
      sphere1.radius + sphere2.radius;
};


/**
 * @param {number} x1
 * @param {number} y1
 * @param {number} z1
 * @param {number} x2
 * @param {number} y2
 * @param {number} z2
 * @return {number}
 */
smash.math.dot = function(x1, y1, z1, x2, y2, z2) {
  return x1 * x2 + y1 * y2 + z1 * z2;
};
